kernel/src/hil/text_screen.rs

```diff
@@ -0,0 +1,99 @@
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The write command is valid and will be sent to the driver.
+    /// - `EBUSY`: The driver is busy with another command.
+    fn print(&self, buffer: &'static mut [u8], len: usize) -> ReturnCode;
+
+    /// Sends to the driver a command to set the cursor at a given position
+    /// (x_position, y_position). When finished, the driver will call the
+    /// `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn set_cursor(&self, x_position: usize, y_position: usize) -> ReturnCode;
+
+    /// Sends to the driver a command to hide the cursor. When finised,
+    /// the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn hide_cursor(&self) -> ReturnCode;
+
+    /// Sends to the driver a command to show the cursor. When finised,
+    /// the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn show_cursor(&self) -> ReturnCode;
+
+    /// Sends to the driver a command to turn on the blinking cursor. When finised,
+    /// the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn blink_cursor_on(&self) -> ReturnCode;
+
+    /// Sends to the driver a command to turn off the blinking cursor. When finised,
+    /// the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn blink_cursor_off(&self) -> ReturnCode;
+
+    /// Sends to the driver a command to turn on the display of the screen.
+    /// When finised, the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn display_on(&self) -> ReturnCode;
+
+    /// Sends to the driver a command to turn off the display of the screen.
+    /// When finised, the driver will call the `command_complete()` callback.
+    ///
+    /// Return values:
+    /// - `SUCCESS`: The command is valid and will be sent to the driver.
+    /// - `EBUSY`: Another command is in progress.
+    fn display_off(&self) -> ReturnCode;
+
+    /// Sends to the driver a command to cleat the display of the screen.
+    /// When finised, the driver will call the `command_complete()` callback.
```

Comment:
Does the cursor reset on a clear?

Reply:
Yes. And everything on the screen will be flushed.